multiraft/multiraft.go

```diff
@@ -644,10 +644,7 @@ func (s *state) start() {
 	})
 }
 
-func (s *state) removePending(g *group, prop *proposal, err error) {
-	if prop == nil {
-		return
-	}
+func (s *state) removePending(g *group, prop proposal, err error) {
```

Comment:
I'm not sure; the log messages are slightly different and I think "cannot remove unknown node 1" is a consequence of this change. But yes, a more direct and deterministic test of this functionality would be valuable.